src/foam/box/SessionServerBox.java

```diff
@@ -88,6 +88,14 @@ public void send(Message msg) {
         if ( user != null ) {
           Group group = (Group) x.get("group");
 
+          if ( authenticate_ && ! auth.check(session.getContext(), "service." + spec.getName()) ) {
+            logger.debug("missing permission", group != null ? group.getId() : "NO GROUP" , "service." + spec.getName());
+            msg.replyWithException(new NoPermissionException("No permission"));
+            return;
+          }
+
+          // padding this cause if group is null this can cause an NPE
+          // technically the user shouldn't be created without a group
```

Comment:
What does "padding this cause" mean?

I think that the group may not be set if the user is acting as an agent.

@TW80000, why is that?

Reply:
@shreymahendru wrote that comment originally in https://github.com/foam-framework/foam2/commit/62795f6ae94f8ad6e4b94d61229089a87a7e8e53. I think "padding this cause" means "wrapping this in a null check because".

> I think that the group may not be set if the user is acting as an agent.

That shouldn't be true. When you act as a user we set the group in the context to the junction group. See https://github.com/foam-framework/foam2/blob/master/src/foam/nanos/auth/AgentUserAuthService.java#L89